apps/prairielearn/src/ee/pages/instructorAiGenerateDraftEditor/instructorAiGenerateDraftEditor.html.ts

```diff
@@ -89,76 +102,90 @@ export function InstructorAiGenerateDraftEditor({
+                    </li>
+                    <li class="nav-item">
+                      <a a class="nav-link" data-toggle="tab" href="#question-code">Files</a>
+                    </li>
+                  </ul>
+                  <div
+                    class="d-flex align-items-center justify-content-end flex-grow-1 border-bottom pr-2"
+                  >
+                    <span data-toggle="modal" data-target="#finalizeModal">
+                      <button
+                        type="button"
+                        class="btn btn-sm btn-primary"
+                        data-toggle="tooltip"
+                        title="Finalize a question to use it on assessments and make manual edits"
+                      >
+                        <i class="fa fa-check" aria-hidden="true"></i>
+                        Finalize question
+                      </button>
+                    </span>
                   </div>
                 </div>
+                <div class="app-preview-content">
+                  ${QuestionAndFilePreview({ resLocals, prompts })}
+                </div>
               </div>
-            </div>
+            </main>
           </div>
-          ${FinalizeModal({ csrfToken: resLocals.__csrf_token })}
-        </div>
-        <div class="card-footer">
-          <a href="#" role="button" data-toggle="modal" data-target="#finalizeModal"
-            >Finalize this question</a
-          >
-          to use it on assessments and make manual edits.
         </div>
-      </div>
-    `,
-  });
+        ${FinalizeModal({ csrfToken: resLocals.__csrf_token })}
+      </body>
+      <script>
+        // TODO: something different on narrow viewports?
+        const chatHistory = document.querySelector('.app-chat-history');
+        chatHistory.scrollTop = chatHistory.scrollHeight;
+      </script>
+    </html>
+  `.toString();
 }
 
 function PromptHistory({
   prompts,
   urlPrefix,
+  showJobLogs,
 }: {
   prompts: AiQuestionGenerationPrompt[];
   urlPrefix: string;
+  showJobLogs: boolean;
 }) {
   return prompts
     .filter((prompt) => prompt.prompt_type !== 'auto_revision')
     .map((prompt) => {
+      // TODO: Once we can upgrade to Bootstrap 5.3, we can use the official
+      // `bg-secondary-subtle` class instead of the custom styles here.
       return html`<div class="d-flex flex-row-reverse">
-          <div class="p-3 mb-2 bg-secondary text-white rounded">${prompt.user_prompt}</div>
+          <div class="p-3 mb-2 rounded" style="background: #e2e3e5; max-width: 90%">
+            ${prompt.user_prompt}
+          </div>
         </div>
         <div class="d-flex flex-row">
-          <div class="p-3 mb-2 bg-dark text-white rounded">
+          <div class="p-3 mb-2 rounded" style="max-width: 90%">
             ${prompt.prompt_type === 'initial'
               ? 'We generated a potential question.'
               : 'We have made changes. Please check the preview and prompt for further revisions.'}
```

Comment:
I feel like the first of these should also have the second sentence ("Please check .. and prompt ...")?

Could we also add "Once you are happy with the question, click the Finalize button." or something like that? I'm still worried that the finalized button is a long way away from the chat flow and it's also not a standard name that people will necessarily recognize.

If we are worried about this message getting too long, we could have the two strings be something like:
```suggestion
              ? 'We generated a potential question. Please check the preview and prompt for further revisions. Once you are happy with the question, click the Finalize button in the top right.'
              : 'We have made changes. Please prompt further or click Finalize.'}
```

Reply:
Just to clarify above, my thinking was to have the first message from the AI be long and explain everything in detail, and then future messages can be short and just a reminder.

Reply:
I like the idea of putting the longer message first. Here's what I came up with:

<img width="377" alt="Screenshot 2025-02-04 at 13 40 55" src="https://github.com/user-attachments/assets/3abab643-2f46-43e2-8554-1f47528d6970" />

Key differences:

- I used the passive voice, it felt weird to talk about "we". If we're going to use a pronoun like that, I'd rather it be "I" like ChatGPT does, but that also feels a little off for us.
- I avoided talking about "click in the upper right" since that breaks on mobile and that's meaningless for folks using screen readers. The "finalize" button is the only primary button on the page besides in the question preview, so I think people should be able to find it easily. Of course, we'll see what happens during user testing.

Reply:
Looks great!